### PR TITLE
Feature/relantionship init

### LINKED ISSUE #12: Init entities and relationships
Every entity that conform to the protocol (ATM called `KakapoSerializable`) will have to implement `init(id: String)`. To create entities there might be 2 ways:

``` swift
db.create(20, User) //pseudocode
// or
db.create(20) { (id) in
    return User(name: Faker.name, friends: db.findRandom(Friends))
}
```

Friends is a relationship od User:

``` swift
struct Friend: KakapoModel {
    let id: String
    let name: String
    init(id: String) {
       self.id = id
       name = Faker.name
    }
}
```

User will init is relationship as:

``` swift
struct User: KakapoModel {
    let id: String
    let name: String
    let friends: [Friend]
    init(id: String) {
       self.id = id
       name = Faker.name
       friends = 1...(arc4random()).each{ Friend(id: NSUUID()) }
    }
}
```

### PR DESCRIPTION
**Pull request by MP0w**
Closes #12

### CODE REVIEW

**Comment by MP0w** [comment]
- [x] Add test for create inside create / insert inside insert
- [x] Fix deadlock for create inside create

**Comment by MP0w** [comment]
Same deadlock for all the synchronous stuff: e.g. find inside a create

File: Source/KakapoDB.swift [hunk 1]
```diff
@@ -34,42 +34,71 @@ private final class ArrayBox<T> {
...
+
+    public init() {
+        dispatch_queue_set_specific(queue, queueKey, queueContext, nil)
+    }
+    
+    private func synchronizeDBWrite(closure: () -> ()) {
+        if dispatch_get_specific(queueKey) == queueContext {
+            closure()
+        } else {
+            dispatch_barrier_sync(queue, closure)
```
Comment (joanromano) [inline Source/KakapoDB.swift]: Can this be async now?

File: KakapoExample/KakapoTests/KakapoDBTests.swift [hunk 2]
```diff
@@ -34,7 +34,7 @@ struct CommentFactory: Storable {
     let likes: Int
     let id: Int
     
-    init(id: Int) {
+    init(id: Int, db: KakapoDB) {
```
Comment (joanromano) [inline KakapoExample/KakapoTests/KakapoDBTests.swift:37]: Shouldn't we use the db here to test the deadlocks as well?

**Comment by joanromano** [comment]
:+1:

**Pull request merged**